api/src/main/java/com/pip/chatbot/controller/CitiesController.java

```diff
@@ -1,86 +1,46 @@
 package com.pip.chatbot.controller;
 
 import com.pip.chatbot.model.City;
-import com.pip.chatbot.repository.CitiesRepository;
+import com.pip.chatbot.payload.response.Response;
+import com.pip.chatbot.payload.response.ResponseStatus;
+import com.pip.chatbot.service.CityService;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
-import java.util.HashMap;
+import java.util.List;
+import java.util.Optional;
 
 @RestController
 public class CitiesController {
-    private final CitiesRepository citiesRepository;
+    private final CityService cityService;
 
-    CitiesController(CitiesRepository citiesRepository) {
-        this.citiesRepository = citiesRepository;
+    CitiesController(CityService cityService) {
+        this.cityService = cityService;
     }
 
-    @GetMapping("/admin/cities")
-    public ResponseEntity<?> getCitiesList() {
-        try {
-            return ResponseEntity.ok(citiesRepository.getAllCities());
-        } catch (Exception e) {
-            return ResponseEntity
-                    .status(400)
-                    .body(new HashMap.SimpleEntry<>("message", "error"));
-        }
+    @GetMapping("/cities")
+    public ResponseEntity<List<City>> getCities(@RequestParam Optional<String> country) {
+        List<City> cities;
```

Comment:
Sprawdzić czy to tak rzeczywiście działa, tzn. że Optional jest pusty dla ścieżki cities i pełny dla ścieżki cities?country=xxx

Reply:
Sprawdzone już przed wysłaniem :)